state/protocol/badger/snapshot_test.go

```diff
@@ -418,7 +421,39 @@ func TestSealingSegment(t *testing.T) {
 			require.Len(t, segment.ExecutionResults, 1)
 		})
 	})
+
+	t.Run("sealing segment where highest block in segment does not seal lowest", func(t *testing.T) {
+		util.RunWithFollowerProtocolState(t, rootSnapshot, func(db *badger.DB, state *bprotocol.FollowerState) {
+			// build a block to seal
+			block1 := unittest.BlockWithParentFixture(head)
+			err := state.Extend(context.Background(), &block1)
+			require.NoError(t, err)
+
+			// build a block sealing block1
+			block2 := unittest.BlockWithParentFixture(block1.Header)
+			receipt1, seal1 := unittest.ReceiptAndSealForBlock(&block1)
+			block2.SetPayload(unittest.PayloadFixture(unittest.WithReceipts(receipt1), unittest.WithSeals(seal1)))
```

Comment:
Could we add an extra block to make this a valid chain segment (can't have receipt and seal for same block, in the same block).
```
A <- B(RA) <- C(SA) <- D
```

Reply:
https://github.com/onflow/flow-go/pull/1533/commits/ef3887b8355cf2913aa1911d598883538769238a